drivers/common/virtio_ha/virtio_ha.c

```diff
@@ -875,7 +876,9 @@ virtio_ha_vf_ctx_query(struct virtio_dev_name *vf,
 		return -1;
 	}
 
-	mem = (struct virtio_vdpa_dma_mem *)msg->iov.iov_base;
+	vfd_saved = (bool *)msg->iov.iov_base;
```

Comment:
this seems not reasonable, can we define a new struct , include bool *vfd_saved; and struct virtio_vdpa_dma_mem *mem;
after query, copy entire struct?